### DISCUSSION TITLE
Allow microsecond/nanosecond precision on span start time and duration

### DISCUSSION THREAD

**Discussion by vjsamuel** [grafana/grafana#45182] [Observability]
The tracing spec on OTEL allows up to nanosecond precision on the start time and duration as per:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#timestamp

How ever the trace visualizer only shows up to the second on the trace start time:

![image](https://user-images.githubusercontent.com/3927228/153279897-ad132b6b-0529-4c2c-838e-a689076f6c6b.png)

Would it be possible to somehow automatically increase the precision in such cases where the precision needs to be higher?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
